### PR TITLE
Support for Azure Blob filesystem update

### PR DESCRIPTION
**Pull request by edd3x**
[Description of PR]
Files  for Azure Blob Filesystem Data Store
The filesystem currently supported by xcube v0.13.0 include "file", "memory, and "AWS S3". This PR comprises updates that enable xcube data store instances to access files from the Microsoft Azure Blob storage filesystem using 
 'abfs' as store id, blob container as root, account_name and account_key or connection_string as store
 params. This is achieved via the <code>fsspec.AbstractFileSystem</code>  using the [fsspec adlfs](https://github.com/fsspec/adlfs) package.

Checklist:
* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in `docs/source/*`
* [x] Changes documented in `CHANGES.md`
* [x] AppVeyor CI passes
* [ ] Test coverage remains or increases (target 100%)

### CODE REVIEW

**Comment by forman** [comment]
@edd3x thanks! Before we advance, could you please describe the PR (description is missing entirely) and tick the checklist items that are addressed by this PR. We will start reviewing only if checklist is complete.

**Comment by edd3x** [comment]
@forman I have added the description to the PR. I ticked some items I have completed on the checklist. Please let me know if you require any other change

**Comment by forman** [comment]
Hi @edd3x, your PR is not forgotten, but I only have limited time this week. Will take care asap.

File: environment.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@ dependencies:
   - rfc3339-validator >=0.1  # for python-jsonschema date-time format validation
   - rioxarray >=0.11
   - s3fs >=2021.6
+  - adlfs >=2023.1 # for azure blob filesystem
```
Comment (forman) [inline environment.yml:39]:
```suggestion
  - adlfs >=2023.1  # for azure blob filesystem
```

PEP8 wants 2 spaces between code and comment (we apply it for YAML as well)

File: test/test_AzureBlobFS.ipynb [hunk 2]
```diff
@@ -0,0 +1,79 @@
+{
```
Comment (forman) [inline test/test_AzureBlobFS.ipynb:1]: Please adapt your Notebook to the conventions used in `example/notebooks/datastores` and move it there. Use the same naming convention.

File: CHANGES.md [hunk 3]
```diff
@@ -2,6 +2,46 @@
...
+        connection_string: xxxxxxxx
+    Datasets:
+      - Path: "*.levels"
+        Style: default
+
+* Added Notebook
+  [test_AzureBlobFS.ipynb](/test/test_AzureBlobFS.ipynb). This notebook shows how a
+  new data store instance can connect and list zarr files from Azure bolb storage using 
+  'abfs' as store id, blob container as root, account_name and account_key or connection_string as store
+  params.
```
Comment (forman) [inline CHANGES.md:43]: See below: Please adapt your Notebook to the conventions used in `example/notebooks/datastores` and move it there. Use the same naming convention.

Mention its new location here.

File: xcube/core/store/fs/impl/fs.py [hunk 4]
```diff
@@ -136,3 +136,21 @@ def get_storage_options_schema(cls) -> JsonObjectSchema:
...
+
+    @classmethod
+    def get_storage_options_schema(cls) -> JsonObjectSchema:
+
+        return JsonObjectSchema(
+            properties=dict(
+                anon=JsonBooleanSchema(
+                    title='Whether to anonymously connect to Azure Blob Storage.'
+                ),
+                **COMMON_STORAGE_OPTIONS_SCHEMA_PROPERTIES,
```
Comment (forman) [inline xcube/core/store/fs/impl/fs.py:153]: Please specify the Azure-specific store parameters here:

* `account_name`
* `account_key`
* `connection_string`

**Review by forman — CHANGES_REQUESTED** [review]
Hey @edd3x, almost done :)

Please see my comments and suggestions. If you agree and fix stuff accordingly, I am happy to quickly approve & merge.

File: test/test_AzureBlobFS.ipynb (hunk 2 above)
Comment (forman) [reply test/test_AzureBlobFS.ipynb:1]: It should be named `8_azure_blob filesystem.ipynb`.

**Review by forman — APPROVED** [review]
Thanks, very good!

**Comment by edd3x** [comment]
Hi @forman, I hope you are well; I have seen the new release of `xcube` and `xcube viewer`. I tried updating my server with the new version `xcube` and got an error when I ran the server. I noticed it was because of this line of code (`**COMMON_STORAGE_OPTIONS_SCHEMA_PROPERTIES,`) on line 172 of the `AzureFsAccessor()` class. 

Commenting or removing that line solved the problem.  Do I have to create a new pull request to fix this or it can be fixed from your end? 
Thanks

**Comment by forman** [comment]
Hi @edd3x 

> Commenting or removing that line solved the problem. Do I have to create a new pull request to fix this

No, I'll have a look and try fixing this immediately. Will release a 1.0.2 asap.

**Pull request merged**